Modules/xSharePoint/DSCResources/MSFT_xSPPasswordChangeSettings/MSFT_xSPPasswordChangeSettings.psm1

```diff
@@ -0,0 +1,85 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)][System.String] $MailAddress,
+        [parameter(Mandatory = $false)][ValidateRange(0,356)][System.UInt32] $DaysBeforeExpiry,
+        [parameter(Mandatory = $false)][ValidateRange(0,36000)][System.UInt32] $PasswordChangeWaitTimeSeconds,
+        [parameter(Mandatory = $false)][ValidateRange(0,99)][System.UInt32] $NumberOfRetries
+    )
+
+    Write-Verbose -Message "Retrieving farm wide automatic password change settings"
+
+    $result = Invoke-xSharePointCommand -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        $farm = Get-SPFarm
+        if ($null -eq $farm ) { return $null }
+        return @{
+            MailAddress = $farm.PasswordChangeEmailAddress
+            PasswordChangeWaitTimeSeconds= $farm.PasswordChangeGuardTime
+            NumberOfRetries= $farm.PasswordChangeMaximumTries
+            DaysBeforeExpiry = $farm.DaysBeforePasswordExpirationToSendEmail 
+        }
+    }
+    return $result
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)][System.String] $MailAddress,
+        [parameter(Mandatory = $false)][ValidateRange(0,356 )][System.UInt32] $DaysBeforeExpiry,
+        [parameter(Mandatory = $false)][ValidateRange(0,36000)][System.UInt32] $PasswordChangeWaitTimeSeconds,
+        [parameter(Mandatory = $false)][ValidateRange(0,99 )][System.UInt32] $NumberOfRetries        
+    )
+
+    Write-Verbose -Message "Updating farm wide automatic password change settings"
+    Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        $farm = Get-SPFarm
+
+        if ($null -eq $farm ) { return $null }
```

Comment:
Can we throw an exception here instead of just returning null so that a meaningful error is reported to the logs?